packages/ckeditor5-clipboard/tests/dragdropexperimental.js

```diff
@@ -0,0 +1,2241 @@
+ */
+
+/* globals document, Event */
+
+import ClipboardPipeline from '../src/clipboardpipeline';
+import DragDropExperimental from '../src/dragdropexperimental';
+import DragDropTarget from '../src/dragdroptarget';
+import PastePlainText from '../src/pasteplaintext';
+
+import Widget from '@ckeditor/ckeditor5-widget/src/widget';
+import WidgetToolbarRepository from '@ckeditor/ckeditor5-widget/src/widgettoolbarrepository';
+import ClassicTestEditor from '@ckeditor/ckeditor5-core/tests/_utils/classictesteditor';
+import Paragraph from '@ckeditor/ckeditor5-paragraph/src/paragraph';
+import Table from '@ckeditor/ckeditor5-table/src/table';
+import HorizontalLine from '@ckeditor/ckeditor5-horizontal-line/src/horizontalline';
+import ShiftEnter from '@ckeditor/ckeditor5-enter/src/shiftenter';
+import BlockQuote from '@ckeditor/ckeditor5-block-quote/src/blockquote';
+import Bold from '@ckeditor/ckeditor5-basic-styles/src/bold';
+import env from '@ckeditor/ckeditor5-utils/src/env';
+
+import testUtils from '@ckeditor/ckeditor5-core/tests/_utils/utils';
+import { getData as getModelData, setData as setModelData } from '@ckeditor/ckeditor5-engine/src/dev-utils/model';
+import { getData as getViewData, stringify as stringifyView } from '@ckeditor/ckeditor5-engine/src/dev-utils/view';
+
+describe( 'Drag and Drop experimental', () => {
+	let editorElement, editor, model, view, viewDocument, root, mapper, domConverter;
+
+	testUtils.createSinonSandbox();
+
+	it( 'requires DragDropTarget, ClipboardPipeline and Widget', () => {
+		expect( DragDropExperimental.requires ).to.deep.equal( [ ClipboardPipeline, Widget, DragDropTarget ] );
+	} );
+
+	it( 'has proper name', () => {
+		expect( DragDropExperimental.pluginName ).to.equal( 'DragDropExperimental' );
+	} );
+
+	it( 'should be disabled on Android', async () => {
+		env.isAndroid = true;
+
+		const editorElement = document.createElement( 'div' );
+		document.body.appendChild( editorElement );
+
+		const editor = await ClassicTestEditor.create( editorElement, {
+			plugins: [ DragDropExperimental ]
+		} );
+
+		const plugin = editor.plugins.get( 'DragDropExperimental' );
+
+		expect( plugin.isEnabled ).to.be.false;
+
+		await editor.destroy();
+		await editorElement.remove();
+
+		env.isAndroid = false;
+	} );
+
+	describe( 'dragging', () => {
+		beforeEach( async () => {
+			editorElement = document.createElement( 'div' );
+			document.body.appendChild( editorElement );
+
+			editor = await ClassicTestEditor.create( editorElement, {
+				plugins: [ DragDropExperimental, PastePlainText, Paragraph, Table, HorizontalLine, ShiftEnter, BlockQuote, Bold ]
+			} );
+
+			model = editor.model;
+			root = model.document.getRoot();
+			mapper = editor.editing.mapper;
+			view = editor.editing.view;
+			viewDocument = view.document;
+			domConverter = view.domConverter;
+		} );
+
+		afterEach( async () => {
+			await editor.destroy();
+			await editorElement.remove();
```

Comment:
Why is void expression awaited?

Reply:
It was copied from dragdrop.js

Reply:
Okay, so without `await` there are memory leaks in tests, so that `await` has a reason.

Reply:
Basing on your fixes of my other comments, I think, you'd removed `await` from `editor.destroy()` and you got memory leaks. My comments only relate to `editorElement.remove()`, because this is not async action. Is it the case?